wallet-api/wallet_rpc.json

```
@@ -0,0 +1,677 @@
+            }
+          }
+        }
+      ],
+      "result": {
+        "name": "result",
+        "description": "The result of the transaction submission",
+        "schema": {
+          "type": "object",
+          "properties": {
+            "transaction_hash": {
+              "title": "Transaction Hash",
+              "$ref": "#/components/schemas/TXN_HASH"
+            }
+          },
+          "required": ["transaction_hash"]
+        }
+      },
+      "errors": [
+        {
+          "$ref": "#/components/errors/INVALID_REQUEST_PAYLOAD"
+        },
+        {
+          "$ref": "#/components/errors/USER_REFUSED_OP"
+        },
+        {
+          "$ref": "#/components/errors/UNKNOWN_ERROR"
+        }
+      ]
+    },
+    {
+      "name": "wallet_addDeclareTransaction",
+      "summary": "Submit a declare transaction",
+      "params": [
+        {
+          "name": "declare_transaction",
+          "description": "The information needed to declare a new class",
+          "required": true,
+          "schema": {
+            "$ref": "#/components/schemas/DECLARE_TXN"
+          }
+        }
+      ],
+      "result": {
+        "name": "result",
+        "description": "The result of the transaction submission",
+        "schema": {
+          "type": "object",
+          "properties": {
+            "transaction_hash": {
+              "title": "The hash of the declare transaction",
+              "$ref": "#/components/schemas/TXN_HASH"
+            },
+            "class_hash": {
+              "title": "The hash of the declared class",
+              "$ref": "#/components/schemas/PADDED_FELT"
+            }
+          },
+          "required": ["transaction_hash", "class_hash"]
+        }
+      },
+      "errors": [
+        {
+          "$ref": "#/components/errors/INVALID_REQUEST_PAYLOAD"
+        },
+        {
+          "$ref": "#/components/errors/USER_REFUSED_OP"
+        },
+        {
+          "$ref": "#/components/errors/UNKNOWN_ERROR"
+        }
+      ]
+    },
+    {
+      "name": "wallet_signTypedData",
+      "summary": "Sign typed data using the wallet",
+      "params": [
+        {
+          "name": "typed_data",
+          "description": "The typed data to sign",
+          "required": true,
+          "schema": {
+            "$ref": "#/components/schemas/TYPED_DATA"
+          }
+        }
+      ],
+      "result": {
+        "name": "response",
+        "summary": "The signature over the typed data",
+        "required": true,
+        "schema": {
+          "$ref": "#/components/schemas/SIGNATURE"
+        }
+      },
+      "errors": [
+        {
+          "$ref": "#/components/errors/INVALID_REQUEST_PAYLOAD"
+        },
+        {
+          "$ref": "#/components/errors/USER_REFUSED_OP"
+        },
+        {
+          "$ref": "#/components/errors/UNKNOWN_ERROR"
+        }
+      ]
+    }
+  ],
+  "components": {
+    "contentDescriptors": {},
+    "schemas": {
+      "TXN_HASH": {
```

Should we call this padded TX hash to differentiate from api TXN_HASH which is based on FELT ?
If agreed it needs to be renamed on usage lines.

```suggestion
      "PADDED_TXN_HASH": {
```